test/SparqlAntlrParserTest.cpp

```diff
@@ -855,3 +855,102 @@ TEST(SparqlParser, GroupClause) {
                              IsExpressionGroupKey("COUNT(?baz)"));
   }
 }
+
+TEST(SparqlParser, propertyPaths) {
+  using Op = PropertyPath::Operation;
+  // Test all the base cases.
+  {
+    // "a" is a special case. It is a valid PropertyPath.
+    // It is short for "<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>".
+    expectCompleteParse(
+        parseVerbPathOrSimple("a", {}),
+        IsPropertyPath(PropertyPath(
+            Op::IRI, 0, "<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>",
+            {})));
+  }
+  { EXPECT_THROW(parseVerbPathOrSimple("b", {}), std::runtime_error); }
+  {
+    expectCompleteParse(
+        parseVerbPathOrSimple("test:foo", {}),
```

Comment:
Does the parser currently not throw on undefined prefixes?
These should really be expanded, you should call `parseIri` or what ever it is called and not `getText()`. We need to get rid of the delayed expansion of the prefixes. Also this means that you need the prefixMap in the `parseWithAntlr`...

Reply:
Not here. I am currently also working on the `valuesClause`. There I'm getting Exceptions on undefined prefixes. I will investigate this.